main/src/domain/targets/track_peak_target.rs

```diff
@@ -60,9 +60,17 @@ impl<'a> Target<'a> for TrackPeakTarget {
 impl TrackPeakTarget {
     fn peak(&self) -> Option<Volume> {
         let reaper = Reaper::get().medium_reaper();
-        let channel_count = unsafe {
-            reaper.get_media_track_info_value(self.track.raw(), TrackAttributeKey::Nchan) as i32
+        
+        let vu_mode = unsafe {
+            reaper.get_media_track_info_value(self.track.raw(), TrackAttributeKey::VuMode) as i32
         };
+        let mut channel_count = 2;
```

Comment:
This would be the more Rusty way to do it:

```suggestion
        let channel_count = if matches!(vu_mode, 2 | 8) {
            unsafe {
                reaper.get_media_track_info_value(self.track.raw(), TrackAttributeKey::Nchan) as i32
            }
        } else {
            2
        };
```

Reply:
Thanks. I've been through the Rust book recently and that definitely looks better. Added an update.